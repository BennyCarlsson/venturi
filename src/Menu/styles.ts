import { NavLink } from 'react-router-dom'
import styled from 'styled-components'
import Colors from 'tokens/colors'

export const MenuWrapper = styled.nav`
  display: flex;
  align-items: center;
  height: 64px;
  background-color: ${Colors.vBlack};
`

export const StyledLink = styled(NavLink)`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: 14px;
  color: ${Colors.white};
  text-decoration: none;

  &.active {
    font-weight: 600;
  }
`

export const SpanOne = styled.span`
  margin-top: 5px;
`
export const SpanTwo = styled.span`
  margin-top: 7px;
`

export const SpanThree = styled.span`
  margin-top: 5px;
`
