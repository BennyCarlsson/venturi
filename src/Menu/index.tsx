import { ReactComponent as BookLogo } from "assets/icons/book-open.svg";
import { ReactComponent as TrainCar } from "assets/icons/train-car.svg";
import MenuDotsIcon from "components/MenuDotsIcon";
import { MenuWrapper, SpanOne, SpanThree, SpanTwo, StyledLink } from "./styles";

const Menu = () => {
  return (
    <MenuWrapper>
      <StyledLink exact to="/" activeClassName="active">
        <BookLogo />
        <SpanOne>Current table</SpanOne>
      </StyledLink>
      <StyledLink to="/departures" activeClassName="active">
        <TrainCar />
        <SpanTwo>Departures</SpanTwo>
      </StyledLink>
      <StyledLink to="/options" activeClassName="active">
        <MenuDotsIcon />
        <SpanThree>Options</SpanThree>
      </StyledLink>
    </MenuWrapper>
  );
};

export default Menu;
