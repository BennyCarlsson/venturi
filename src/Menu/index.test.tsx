import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Menu from '.'
import { BrowserRouter as Router } from 'react-router-dom'
import userEvent from '@testing-library/user-event'

describe('<Menu />', () => {
  beforeEach(() => {
    render(
      <Router>
        <Menu />
      </Router>
    )
  })

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Menu />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    expect(screen).toBeTruthy()
  })

  it('should render all 3 links', () => {
    expect(screen.getAllByRole('link')).toHaveLength(3)
  })

  it('should go away from startscreen and then back', () => {
    expect(window.location.pathname).toBe('/')
    userEvent.click(screen.getByText('Options'))
    expect(window.location.pathname).not.toBe('/')
    userEvent.click(screen.getByText('Current table'))
    expect(window.location.pathname).toBe('/')
  })

  it('should start on start screen', () => {
    expect(window.location.pathname).toBe('/')
  })

  it('should go to Departures screen', () => {
    userEvent.click(screen.getByText('Departures'))
    expect(window.location.pathname).toBe('/departures')
  })

  it('should go to Options screen', () => {
    userEvent.click(screen.getByText('Options'))
    expect(window.location.pathname).toBe('/options')
  })
})
