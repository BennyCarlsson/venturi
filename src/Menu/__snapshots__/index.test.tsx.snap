// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match snapshot 1`] = `
<nav
  className="sc-dlfnbm ebxcBV"
>
  <a
    aria-current="page"
    className="sc-hKgILt cRuwcv active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <svg>
      book-open.svg
    </svg>
    <span
      className="sc-eCssSg kcQKoW"
    >
      Current table
    </span>
  </a>
  <a
    aria-current={null}
    className="sc-hKgILt cRuwcv"
    href="/departures"
    onClick={[Function]}
  >
    <svg>
      train-car.svg
    </svg>
    <span
      className="sc-jSgupP dxonMR"
    >
      Departures
    </span>
  </a>
  <a
    aria-current={null}
    className="sc-hKgILt cRuwcv"
    href="/options"
    onClick={[Function]}
  >
    <div
      className="sc-bdfBwQ hvQWz"
    >
      <div
        className="sc-gsTCUz hslkzR"
      />
      <div
        className="sc-gsTCUz hslkzR"
      />
      <div
        className="sc-gsTCUz hslkzR"
      />
      <div
        className="sc-gsTCUz hslkzR"
      />
      <div
        className="sc-gsTCUz hslkzR"
      />
      <div
        className="sc-gsTCUz hslkzR"
      />
      <div
        className="sc-gsTCUz hslkzR"
      />
      <div
        className="sc-gsTCUz hslkzR"
      />
      <div
        className="sc-gsTCUz hslkzR"
      />
    </div>
    <span
      className="sc-gKsewC jKkyBy"
    >
      Options
    </span>
  </a>
</nav>
`;
